### PR TITLE
MGMT-1837 Reduce timeouts of hosts during installation, according to the step they are in

### PR DESCRIPTION
**Pull request by oshercc**

### CODE REVIEW

File: internal/host/transition.go [hunk 1]
```diff
@@ -409,6 +410,18 @@ func (th *transitionHandler) HasClusterError(sw stateswitch.StateSwitch, args st
 	return swag.StringValue(cluster.Status) == models.ClusterStatusError, nil
 }
 
+func (th *transitionHandler) HasInstallationTimedOut(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) (bool, error) {
+	sHost, ok := sw.(*stateHost)
+	if !ok {
+		return false, errors.New("HasInstallationTimedOut incompatible type of StateSwitch")
+	}
+	maxDuration, ok := InstallationProgressTimeout[sHost.host.Progress.CurrentStage]
```
Comment (eranco74) [inline internal/host/transition.go:418]: There are more stages that are not in this map. 
These are just the ones we have timeout for, please handle the case where the host current stage isn't in the map (return false).

File: internal/host/transition.go [hunk 2]
```diff
@@ -409,6 +410,18 @@ func (th *transitionHandler) HasClusterError(sw stateswitch.StateSwitch, args st
...
+	if !ok {
+		return false, nil
```
Comment (eranco74) [inline internal/host/transition.go:420]: Consider adding debug log

File: internal/host/transition.go (hunk 1 above)
Comment (oshercc) [reply internal/host/transition.go:418]: @eranco74 default 1 hour?

File: internal/host/host.go [hunk 3]
```diff
@@ -51,6 +49,15 @@ var manualRebootStages = [...]models.HostStage{
...
 }
 
+var InstallationProgressTimeout = map[models.HostStage]time.Duration{
+	models.HostStageStartingInstallation: 10 * time.Minute,
+	models.HostStageWritingImageToDisk:   20 * time.Minute,
+	models.HostStageRebooting:            30 * time.Minute,
+	models.HostStageConfiguring:          30 * time.Minute,
+	models.HostStageWaitingForIgnition:   60 * time.Minute,
+	models.HostStageInstalling:           20 * time.Minute,
+}
```
Comment (gamli75) [inline internal/host/host.go:59]: what about the other stages: HostStageJoined, HostStageWaitingForControlPlane, HostStageStartWaitingForControlPlane ?

File: internal/host/transition_test.go [hunk 4]
```diff
@@ -1017,6 +1017,61 @@ var _ = Describe("Refresh Host", func() {
 		hostId = strfmt.UUID(uuid.New().String())
 		clusterId = strfmt.UUID(uuid.New().String())
 	})
+
+	Context("host installation timeout", func() {
```
Comment (gamli75) [inline internal/host/transition_test.go:1021]: please add a test for a stage that is not in the map and verify that it stays in the correct status and another test that verifies that if the time didn't get to the threshold - it stays in the correct status

File: internal/host/statemachine.go [hunk 5]
```diff
@@ -233,6 +233,16 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		PostTransition:   th.PostRefreshHost(statusInfoAbortingDueClusterErrors),
 	})
 
+	// Time out while host installation
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusInstallingInProgress)},
+		Condition:        stateswitch.And(th.HasInstallationTimedOut, stateswitch.Not(th.HasClusterError)),
```
Comment (filanov) [inline internal/host/statemachine.go:241]: why do you need `th.HasClusterError` if cluster is already in error? 
I don't see how cluster can be in error and in progress?

File: internal/host/host.go [hunk 6]
```diff
@@ -51,6 +49,16 @@ var manualRebootStages = [...]models.HostStage{
...
 }
 
+var InstallationProgressTimeout = map[models.HostStage]time.Duration{
+	models.HostStageStartingInstallation: 10 * time.Minute,
+	models.HostStageWritingImageToDisk:   20 * time.Minute,
+	models.HostStageRebooting:            30 * time.Minute,
+	models.HostStageConfiguring:          30 * time.Minute,
+	models.HostStageWaitingForIgnition:   60 * time.Minute,
+	models.HostStageInstalling:           20 * time.Minute,
+	"DEFAULT":                            60 * time.Minute,
```
Comment (filanov) [inline internal/host/host.go:59]: Probably should be a const.

Comment (filanov) [reply internal/host/host.go:59]: Actually i'm not sure that we should have a default. we probably want to fail if we have something unexpected.

File: internal/host/host.go [hunk 7]
```diff
@@ -51,6 +49,16 @@ var manualRebootStages = [...]models.HostStage{
 	models.HostStageDone,
 }
 
+var InstallationProgressTimeout = map[models.HostStage]time.Duration{
+	models.HostStageStartingInstallation: 10 * time.Minute,
+	models.HostStageWritingImageToDisk:   20 * time.Minute,
+	models.HostStageRebooting:            30 * time.Minute,
```
Comment (filanov) [inline internal/host/host.go:55]: Will this timeout work on bare-metal?

**Pull request merged**